### PR TITLE
 thunderbird, thunderbird-bin: 68.2.2 -> 68.3.0 [High security fixes]

### PR DESCRIPTION
**Pull request by taku0**
###### Motivation for this change

- High security fixes.
- Other bug fixes and updates.

https://www.thunderbird.net/en-US/thunderbird/68.3.0/releasenotes/
https://www.mozilla.org/en-US/security/advisories/mfsa2019-38/

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Tested both `thunderbird-bin` and `thunderbird`.

###### Notify maintainers

cc @Fuuzetsu, @nbp, @edolstra

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/thunderbird/default.nix [hunk 1]
```diff
@@ -53,7 +53,7 @@ in stdenv.mkDerivation rec {
     # Remove buildconfig.html to prevent a dependency on clang etc.
     ./no-buildconfig.patch
   ]
-  ++ lib.optional (lib.versionOlder version "69")
+  ++ lib.optional (lib.versionOlder version "68.3.0")
```
Comment (das-g) [inline pkgs/applications/networking/mailreaders/thunderbird/default.nix]: Does this mean that 68.3.0 itself shall not have [this patch](https://hg.mozilla.org/integration/autoland/rev/562655fedb9d) applied?

If so, is it nonetheless best practice to leave that lib.optional clause in the derivation, now that the version _is_ at that level?

**Review by das-g — COMMENTED** [review]
Looks good to me, but I have a question:

File: pkgs/applications/networking/mailreaders/thunderbird/default.nix (hunk 1 above)
Comment (taku0) [reply pkgs/applications/networking/mailreaders/thunderbird/default.nix]: Removed the patch.

**Review by das-g — APPROVED** [review]

**Comment by lovesegfault** [comment]
I'll suggest that this be superseded by #75328.

**Comment by das-g** [comment]
> I'll suggest that this be superseded by #75328.

#75328 makes much more extensive changes than #74973, which might therefore also be more difficult and time-consuming to review and might warrant more testing. So unless these changes are all needed to be able to bump the version without issues, I'd suggest to get in just the version bump _first_ by reviewing and merging #74973 and _then_ focus on the the remaining changes of #75328.

That way, we'd get the open security issues out of the way sooner and we'll be able also easily backport the version bump to the stable 19.09 channel (already ongoing in #75126) to make it more secure, too, without introducing changes too overreaching for a "stable" branch. We could then (or already in parallel) review #75328 without haste and if some changes of #75328 turn out to be controversial, that won't delay getting out the security fixes.

**Comment by lovesegfault** [comment]
@das-g Fair enough, agreed :)

**Comment by lovesegfault** [comment]
This should be closed since https://github.com/NixOS/nixpkgs/pull/75328 was merged.

**Pull request closed without merging**